src/utils/assign.js

```diff
@@ -0,0 +1,11 @@
+export function assign(target, ...sources) {
```

Comment:
Not exactly the same as `{...foo, ...bar}`, but close enough for our purposes I think